### PR TITLE
Move metric No-Op to `metric/noop`

### LINKED ISSUE #3886: Metric No-Op API allows Meter to be created outside of MeterProvider
[The specification states](https://github.com/open-telemetry/opentelemetry-specification/blob/6ec54bcd61f1c162b4457771df52792b01cf69cd/specification/metrics/noop.md#meter):

> The No-Op MUST NOT provide a way for a user to create a Meter other than by a No-Op MeterProvider.

Currently, a no-op `Meter` can be created with `NewNoopMeter`:

https://github.com/open-telemetry/opentelemetry-go/blob/0694868e556fb4a49acf88a67adda9cf8c8e3b5b/metric/noop.go#L35-L38

This needs to be removed for the API to be compliant with the specification.

### PR DESCRIPTION
**Pull request by MrAlias**
Fix #3886: the existing way for a user to create a no-op `Meter` without using the `MeterProvider` is removed.

- Move metric No-Op to its own package
- Export No-Op types

### Move metric No-Op to its own package

The `go.opentelemetry.io/otel/metric` package is intended to primarily be used by instrumentation authors. Having the No-Op implementation in that package, something they will not interact with, only pollutes the API and clutters the documentation for these users.

This API pollution and documentation clutter is compounded when [exporting No-Op types](#export-no-op-types). It would mean there would be ~20 objects in `go.opentelemetry.io/otel/metric` that all some prefix of `Noop`. Instead of using a prefix to collect common objects, they are moved to their own package.

### Export No-Op types

Part of #3865 includes providing No-Op implementations for all the API interfaces. These types can then be embedded in implementations that want to default to No-Op behavior instead of panicking.

What about the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/6ec54bcd61f1c162b4457771df52792b01cf69cd/specification/metrics/noop.md)?

> The No-Op MUST NOT provide a way for a user to create a Meter other than by a No-Op MeterProvider.

> Counters are always created by a Meter, the No-Op MUST NOT provide a way for a user to create a Counter other than by a No-Op Meter.

> UpDownCounters are always created by a Meter, the No-Op MUST NOT provide a way for a user to create a UpDownCounter other than by a No-Op Meter.

> Histograms are always created by a Meter, the No-Op MUST NOT provide a way for a user to create a Histogram other than by a No-Op Meter.

> Asynchronous Counters are always created by a Meter, the No-Op MUST NOT provide a way for a user to create a Asynchronous Counter other than by a No-Op Meter.

> Asynchronous UpDownCounters are always created by a Meter, the No-Op MUST NOT provide a way for a user to create a Asynchronous UpDownCounter other than by a No-Op Meter.

> Asynchronous Gauges are always created by a Meter, the No-Op MUST NOT provide a way for a user to create a Asynchronous Gauge other than by a No-Op Meter.

All of these requirements mean that the No-Op cannot provide a way for a user to create these types accept via the approved creation types.

I think, given the `noop` package does not provide creation functions for anything other than the `MeterProvider`, this approach still satisfies the specification. It true that a user could use the Go language features on their own to create one of these types, but I think the specification would say that the API "must not allow a user to create" these types outside of via the approved creation types.

### CODE REVIEW

**Comment by MrAlias** [comment]
https://github.com/open-telemetry/opentelemetry-specification/pull/3322

**Comment by MrAlias** [comment]
@Aneurysm9: thinking about https://github.com/open-telemetry/opentelemetry-go/issues/3865 last night, I think we don't have to add a private method to all the interfaces, but could still provide this No-Op implementation. That way, an SDK author could choose to:

1. Leave their implementation without an embedded implementation (current default)
2. Embed the interface itself so it would panic on uninitialized methods (not sure why they would do this, but :shrug:)
3. Embed this No-Op implementation in theirs if they want the default behavior to be No-Op

(This still wouldn't change the `Observable` interface being embedded in all observables)

Thoughts?

**Comment by Aneurysm9** [comment]
That works for me as long as the spec stops prohibiting direct construction of no-op implementations.  I still think we need giant flashing warning signs telling SDK creators and users that they risk having silently incorrect implementations if they choose to embed no-op implementations.  It feels like a bad idea to me, but all the other options are also bad.

File: metric/noop/noop.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+}
+
+// Float64ObservableUpDownCounter is an OpenTelemetry ObservableUpDownCounter
+// used to record float64 measurements. It produces no telemetry.
+type Float64ObservableUpDownCounter struct {
+	instrument.Float64ObservableUpDownCounter
+}
+
+// Int64Observer is a recorder of int64 measurements that performs no operation.
+type Int64Observer struct{ instrument.Int64Observer }
```
Comment (pellared) [inline metric/noop/noop.go:260]: Looks like we should remove the embedding of `instrument.Int64Observer`

File: metric/noop/noop.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+// Observe performs no operation.
+func (Int64Observer) Observe(int64, ...attribute.KeyValue) {}
+
+// Float64Observer is a recorder of float64 measurements that performs no
+// operation.
+type Float64Observer struct{ instrument.Float64Observer }
```
Comment (pellared) [inline metric/noop/noop.go:267]: Looks like we should remove the embedding of `instrument.Float64Observer`

File: metric/noop/noop.go (hunk 1 above)
Comment (MrAlias) [reply metric/noop/noop.go]: Updated. PTAL.

File: metric/noop/noop.go (hunk 2 above)
Comment (MrAlias) [reply metric/noop/noop.go]: Updated. PTAL.

**Review by Aneurysm9 — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Pull request merged**